ZEP000-prevent-ood.md

```diff
@@ -0,0 +1,137 @@
+
+Broker starts a `DiskSpaceUsageMonitor` which periodically checks for disk space usage. The frequency at which it monitors is configured by `diskUsageMonitoringInterval`. The monitor notifies registered `DiskSpaceUsageListener`s, when the disk space usage goes above `diskUsageCommandWatermark` and when it goes below again.
+
+On disk space not available the `CommandApiService` goes to "reject all requests" mode, until the listener notifies that the disk space is available again.
+When the disk space is not available `ZeebePartition` pauses the `StreamProcessor` which in turn notifies the `DueDateTimerChecker` and `JobTimeoutTrigger` to stop triggering the timers. This is done by changing the phase of StreamProcessor to `PAUSED`. When the phase is `PAUSED`, StreamProcessor does not process any records, but it will be still running. When the disk space becomes available again, `ZeebePartition` resumes the `StreamProcessor` by changing the phase back to `PROCESSING`.
+Similarly the actors that receives commands (`LeaderManagementRequestHandler` and `SubscriptionApiCommandMessageHandlerService`) from other partition also listens to disk space usage and stop writing more events to the logStream.
+
+The configuration `diskUsageReplicationWatermark` is used by the Journal. When the disk space usage is greater than diskUsageReplicationWatermark, it does not create new segments and throws `OutOfDiskSpace` exception. We recommend `diskUsageCommandWatermark < diskUsageReplicationWatermark` to make sure that the leader can commit all events that are written by the StreamProcessor before rejecting the requests. Otherwise, it can happen that we cannot commit a snapshot because it is waiting for an event to be committed.
+
+Note that we don't trigger fail over when a leader goes out of disk space, because if the follower is also out of disk space it will lead to a situation where we can never recover. On the other hand, if the leader waits until the exporter failure is resolved, it can eventually take a snapshot and compact.
+
+
+- Does the ZEP affect the official Zeebe distribution (e.g. configuration, logging)?
+	- It add new configuration parameters
+- Does the ZEP require coordination with the platform team? - May be - to find an optimal value for the parameters
+- Does the ZEP require coordination with the Operate team? - No
+
+## Compatibility
+
+There are no compatability breaking changes.
+
+## Testing
+
+You should describe what is the overall functionality that should be tested.
+
+If you are omitting tests, explain why, and explain the impact if it fails, specifically the worst case scenario.
+
+In each of the sections below, we should already list known cases that need to be tested in the final implementation, and at which level. The initial version here should be a best of effort: it is perfectly acceptable and expected that this section will be amended during implementation.
+
+### Unit
```

Comment:
I think we can also have unit test for the smaller components, to see/verify that they react correctly on listener call ?